api/nakadi-event-bus-api.yaml

```diff
@@ -606,22 +488,394 @@ paths:
+      parameters:
+        - name: subscription_id
+          in: path
+          type: string
+          description: Id of subscription
+          required: true
+        - name: X-Nakadi-StreamId
+          in: header
+          type: string
+          description: |
+            Id of stream which client uses to read events. It is not possible to make a commit for a terminated or
+            none-existing stream. Also the client can't commit something which was not sent to his stream.
+          required: true
+        - name: cursors
+          in: body
+          description: Cursors with offsets to commit.
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/Cursor'
+          required: true
+      responses:
+        '200':
+          description: Offsets were committed
+        '207':
+          description: |
+            At least one cursor which was tried to be committed is older or equal to already committed one. Array
+            of commit results is returned for this status code.
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/CursorCommitResult'
         '401':
           description: Client is not authenticated
           schema:
             $ref: '#/definitions/Problem'
+        '403':
+          description: Client is not authorized to perform this operation
+          schema:
+            $ref: '#/definitions/Problem'
+        '404':
+          description: Subscription not found
+          schema:
+            $ref: '#/definitions/Problem'
+        '422':
+          description: Unprocessable Entity
+          schema:
+            $ref: '#/definitions/Problem'
+        '500':
+          description: Server error
+          schema:
+            $ref: '#/definitions/Problem'
+        '503':
+          description: Service (temporarily) unavailable
+          schema:
+            $ref: '#/definitions/Problem'
+
+  /subscriptions/{subscription_id}/events:
+    get:
+      tags:
+        - stream-api
+        - managed-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: |
+        Starts a new stream for reading events from this subscription. The data will be automatically rebalanced
+        between streams of one subscription. The minimal consumption unit is a partition, so it is possible to start as
+        many streams as the total number of partitions in event-types of this subscription. The rebalance currently
+        only operates with the number of partitions so the amount of data in event-types/partitions is not considered
+        during autorebalance.
+        The position of the consumption is managed by Nakadi. The client is required to commit the cursors he gets in
+        a stream. This also depends on the `commit_mode` parameter that is set for the stream.
+      parameters:
+        - $ref: '#/parameters/SubscriptionId'
+        - name: commit_timeout
```

Comment:
> , I thought its about server-side resource management. At least that is the only reason I see for having such a timeout.

If that was the case, why allow clients to control it? Also why set it on the subscription,which is static, instead of the connection, or make it adaptive based on the message/commit ratio. I don't think it's here only for server resource reasons, but it can help with that. It can indicate that clients are not checkpointing, the problem is there's no easy way to tell what's going versus servers closing connections simply to manage resources.